ingests/shift.py

```diff
@@ -0,0 +1,357 @@
+    "Asia and Oceania",
+    "Australia",
+    "Austria",
+    "Azerbaijan",
+    "Bahrain",
+    "Bangladesh",
+    "Belarus",
+    "Belgium",
+    "Benin",
+    "Bolivia",
+    "Botswana",
+    "Brazil",
+    "Brunei Darussalam",
+    "Bulgaria",
+    "Burma",
+    "Cambodia",
+    "Cameroon",
+    "Canada",
+    "Central and South America",
+    "Chile",
+    "China",
+    "Colombia",
+    "Congo",
+    "Croatia",
+    "Cuba",
+    "Cyprus",
+    "Denmark",
+    "Dominican Republic",
+    "EU28",
+    "Ecuador",
+    "Egypt",
+    "El Salvador",
+    "Eritrea",
+    "Estonia",
+    "Ethiopia",
+    "Eurasia",
+    "Europe",
+    "Finland",
+    "France",
+    "Gabon",
+    "Georgia",
+    "Germany",
+    "Ghana",
+    "Gibraltar",
+    "Greece",
+    "Guatemala",
+    "Haiti",
+    "Honduras",
+    "Hungary",
+    "Iceland",
+    "India",
+    "Indonesia",
+    "Iraq",
+    "Ireland",
+    "Israel",
+    "Italy",
+    "Ivory Coast",
+    "Jamaica",
+    "Japan",
+    "Jordan",
+    "Kazakhstan",
+    "Kenya",
+    "Kosovo",
+    "Kuwait",
+    "Kyrgyzstan",
+    "Latvia",
+    "Lebanon",
+    "Libya",
+    "Lithuania",
+    "Malaysia",
+    "Malta",
+    "Mauritius",
+    "Mexico",
+    "Middle East",
+    "Mongolia",
+    "Montenegro",
+    "Morocco",
+    "Mozambique",
+    "NZ",
+    "Namibia",
+    "Nepal",
+    "Netherlands",
+    "Nicaragua",
+    "Niger",
+    "Nigeria",
+    "North America",
+    "Norway",
+    "OECD",
+    "OPEC",
+    "Oman",
+    "Pakistan",
+    "Panama",
+    "Paraguay",
+    "Persian Gulf",
+    "Peru",
+    "Poland",
+    "Portugal",
+    "Qatar",
+    "Romania",
+    "Russian Federation & USSR",
+    "Senegal",
+    "Serbia",
+    "Singapore",
+    "Slovakia",
+    "Slovenia",
+    "South Africa",
+    "Spain",
+    "Sudan",
+    "Suriname",
+    "Sweden",
+    "Tajikistan",
+    "Thailand",
+    "Togo",
+    "Tunisia",
+    "Turkey",
+    "Ukraine",
+    "United Kingdom",
+    "United States of America",
+    "Uruguay",
+    "Uzbekistan",
+    "World",
+    "Yemen",
+    "Zambia",
+    "Zimbabwe",
+]
+
+
+def prepare_query_url(energy_source: str, countries: List[str]) -> str:
+    """Prepare a query URL to request data for a specific energy source and a list of countries.
+
+    Parameters
+    ----------
+    energy_source : str
+        Name of energy source (e.g. "coal").
+    countries : list
+        Countries to include in the query.
+
+    Returns
+    -------
+    query_url : str
+        Query URL to use to request data.
+
+    """
+    # Prepare a query url for request.
+    query_url = (
+        f"{SHIFT_URL}energy/{energy_source}?chart-type=line&chart-types=line&chart-types=ranking&"
+        f"disable-en=false&energy-unit={ENERGY_UNIT}"
+    )
+    # Add each country to the url.
+    for country in countries:
+        query_url += f"&group-names={country.replace(' ', '%20').replace('&', '%26')}"
+    # Add some conditions to the query (not all of them may be necessary).
+    query_url += (
```

Comment:
Nitpick: It might be a bit more readable if you put these parameters into `request.get(..., params=`, but don't waste your time by changing this.